provider/oci/oci.go

```diff
@@ -291,7 +292,7 @@ func (p *OCIProvider) ApplyChanges(ctx context.Context, changes *plan.Changes) e
 			ZoneNameOrId:            &zoneID,
 			PatchZoneRecordsDetails: dns.PatchZoneRecordsDetails{Items: ops},
 		}); err != nil {
-			return err
+			return provider.NewSoftError(err)
```

Comment:
I understand the use of `NewSoftError` for Get or List Records / Zones
Here, it's when patching zone record, so I have doubts. 
Can we really be sure it won't have any unexpected side effect ?

Reply:
In this case, the new behavior is calling this a Fatal error right away.  By wrapping this in a SoftError the previous expected behavior is restored.  The new unexpected behavior would be to Fatal here.  Doesn't it make more sense to have the pod in this case retry in an hour rather than crash the pod and have to recreate the Update immediately.  I don't recall seeing external-dns pods being killed when updates failed.

Reply:
Basically, this change to return a Fatal error has already caused a major unexpected failure of external-dns pods across multiple regions in our tenancy.  This has never happened before.  So this change is trying to restore what was previous behavior to avoid getting bit by this again in a slightly different way.